all screens

Dmitriy-2001 opened the pull request:
картинки в png, ибо в svg почему то не работает

Dmitriy-2001 committed:
all screens

Dmitriy-2001 committed:
detekt fix

Secur1ty-21 commented on app/src/main/res/drawable/ic_back.png:
Маленькие  иконки лучше на svg заменить

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/AboutLevelsScreen.kt:
```diff
@@ -1,13 +1,149 @@
+                .fillMaxWidth()
+                .wrapContentHeight()
+                .align(Alignment.TopCenter),
+            cornerRadius = VolleyDimens.DIMEN_24,
```
Тут вроде 32 по дизайну

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/AboutLevelsScreen.kt:
```diff
@@ -1,13 +1,149 @@
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(VolleyDimens.DIMEN_20.dp),
+                horizontalAlignment = Alignment.Start
```
В column по умолчанию start, можно не указывать

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/AboutLevelsScreen.kt:
```diff
@@ -1,13 +1,149 @@
+                        text = stringResource(id = R.string.level_light_description),
+                        color = VolleyColor.White
+                    )
+                }
```
Лучше вынести этот компонент в отдельную функцию для соблюдения DRY + такой элемент можно делать через AnnotedString одним компонентом, пример без стилизации:
```kotlin
Text(
                    text = buildAnnotatedString {
                        withStyle(style = SpanStyle(brush = gradientBrush)) {
                            append(stringResource(id = R.string.level_light_prefix) + " ")
                        }
                        append(stringResource(id = R.string.level_light_description))
                    },
                    color = VolleyColor.White
)
```

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/AboutLevelsScreen.kt:
```diff
@@ -1,13 +1,149 @@
+                            .align(Alignment.CenterStart)
+                            .size(VolleyDimens.DIMEN_24.dp)
+                            .clickable { navController.popBackStack() }
+                    )
```
Кнопку с иконкой лучше делать через существующий библиотечный компонент + добавлять content description
```kotlin
IconButton(
        onClick = { navController.popBackStack() },
        modifier = Modifier
            .align(Alignment.CenterStart)
            .size(VolleyDimens.DIMEN_24.dp)
    ) {
        Icon(
            painter = painterResource(id = R.drawable.ic_back),
            contentDescription = stringResource(R.string.back),
            tint = VolleyColor.White,
            modifier = Modifier.fillMaxSize()
        )
    }
```

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/LaunchScreen.kt:
```diff
@@ -17,10 +32,33 @@ private const val LAUNCH_DELAY_MILLIS = 3_000L
     ) {
-        CircularProgressIndicator()
+        Image(
+            painter = painterResource(id = R.drawable.bg_launch), // твой png фон
```
gpt?)

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/LaunchScreen.kt:
```diff
@@ -17,10 +32,33 @@ private const val LAUNCH_DELAY_MILLIS = 3_000L
+                contentDescription = null,
+                modifier = Modifier.size(200.dp)
+            )
+            Spacer(modifier = Modifier.height(16.dp))
```
В дизайне 24

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/AboutLevelsScreen.kt:
```diff
@@ -1,13 +1,149 @@
+                    .padding(VolleyDimens.DIMEN_20.dp),
+                horizontalAlignment = Alignment.Start
+            ) {
+                Box(modifier = Modifier.fillMaxWidth()) {
```
Good practice было бы сделать через Row, чтобы избежать таких ситуаций 
<img width="500" height="484" alt="{9B124182-3672-42B6-8159-C16854D983D8}" src="https://github.com/user-attachments/assets/b1132a8d-f049-4760-ae8c-87d6f264b0b2" />

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/AboutLevelsScreen.kt:
```diff
@@ -1,13 +1,149 @@
+                    colors = listOf(VolleyColor.YellowForGradient, VolleyColor.GreenForGradient)
+                )
+
+                Row(verticalAlignment = Alignment.Top) {
```
По умолчанию verticalAligment и так Top, можно не указывать

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/AboutLevelsScreen.kt:
```diff
@@ -1,13 +1,149 @@
+                    )
+                }
+
+                Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_16.dp))
```
Ближе к 20 чем к 16

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/AboutLevelsScreen.kt:
```diff
@@ -1,13 +1,149 @@
+                    )
+                }
+
+                Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_20.dp))
```
Вроде лишний отступ

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/OnboardingScreen.kt:
```diff
@@ -1,33 +1,107 @@
+                .padding(horizontal = VolleyDimens.DIMEN_28.dp),
+            horizontalAlignment = Alignment.Start
+        ) {
+            Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_40.dp))
```
Тут будет 84 если учитывать расстояние system ui, но лучше добавить в аргументы paddingValues и вручную проставить где нужно отступ от system ui

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/RegistrationScreen.kt:
Что-то пошло не так)
<img width="573" height="746" alt="{95DAC143-97D0-4FB2-924C-E31E3D070191}" src="https://github.com/user-attachments/assets/e0083050-ba1f-4a91-83ca-37306991e3e2" />

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/SignUpScreen.kt:
```diff
@@ -1,39 +1,161 @@
+                    contentDescription = null,
+                    modifier = Modifier
+                        .size(VolleyDimens.DIMEN_24.dp)
+                        .padding(end = VolleyDimens.DIMEN_12.dp)
```
Тут надо поменять местами чтобы совпадало с дизайном
```kotlin
Modifier.padding(end = VolleyDimens.DIMEN_12.dp)
.size(VolleyDimens.DIMEN_24.dp)
```

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/SignUpScreen.kt:
```diff
@@ -1,39 +1,161 @@
+            Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_28.dp))
+
+            VolleyText.TitleXLAlt(
+                text = stringResource(id = R.string.sign_up),
```
Тут лишний пробел перед "a" в ресурсах

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/SignUpScreen.kt:
```diff
@@ -1,39 +1,161 @@
+                .fillMaxWidth()
+                .align(Alignment.BottomCenter)
+                .background(
+                    color = VolleyColor.TurquoiseDark,
```
Цвет в дизайне указан другой

Secur1ty-21 commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/SignUpScreen.kt:
```diff
@@ -1,39 +1,161 @@
+                .background(
+                    color = VolleyColor.TurquoiseDark,
+                    shape = RoundedCornerShape(
+                        topStart = VolleyDimens.DIMEN_24.dp,
```
В дизайне 32

Secur1ty-21 reviewed (changes requested):
Проделана большая работа, в целом все экраны похожи на то что в дизайне, расхождений много, но возможно дизайнеры что-то меняли после того  как работа была сделана + не увидел экран RegistrationByPhoneScreen который есть в этом flow

Dmitriy-2001 committed:
Архитектура + MVI обвес

Dmitriy-2001 committed:
исправлены ошибки по верстке + картинки в svg

Dmitriy-2001 committed:
ох уж этот detekt..

Dmitriy-2001 committed:
ох уж этот detekt2

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/aboutlevels/AboutLevelsScreen.kt:
```diff
@@ -0,0 +1,166 @@
+                    gradient = gradientBrush
+                )
+
+                Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_20.dp))
```
Этот Spacer не нужен, у тебя уже выставлен padding 20.dp

Commit:
Merge branch 'develop' into 61-screen-flow-authorization

# Conflicts:
#	app/src/main/java/cy/volleybolley/core/presentation/ui/model/VolleyDimens.kt
#	app/src/main/java/cy/volleybolley/core/presentation/ui/navigation/NavHostContainer.kt
#	app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/LaunchScreen.kt
#	app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/RegistrationScreen.kt

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/aboutlevels/AboutLevelsScreen.kt:
```diff
@@ -0,0 +1,166 @@
+            mainContainerAlignment = Alignment.TopStart,
+            contentContainerAlignment = Alignment.TopStart
+        ) {
+            Column(
```
Я бы вынес в отдельный compose весь верхний текст со стрелкой, он у нас много где используется. Проще будет унифицировать

Commit:
#71 Исправление detekt после merge

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/aboutlevels/AboutLevelsScreen.kt:
```diff
@@ -0,0 +1,166 @@
+
+                Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_16.dp))
+
+                val gradientBrush = Brush.verticalGradient(
```
Кажется, градиент не очень корректно отрабатывает. Как будто красит каждое слово по-разному

Secur1ty-21 replied:
Да, тоже заметил. Баг библиотеки с compose, градиент учитывает высоту всего текста, поэтому когда он идет в две линии. то цвет тянется до второй. Должно только градиентить высоту выделенного текста. Самое забавное что у них в доке пример рабочий, выделяет только то что нужно, но в доке есть верхний текст с "\n", если тут добавить надпись с "\n" с верху то тоже магическим образом все ок)

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/launch/LaunchScreen.kt:
```diff
@@ -0,0 +1,87 @@
+                contentDescription = null,
+                modifier = Modifier.size(VolleyDimens.DIMEN_200.dp)
+            )
+            Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_16.dp))
```
Здесь по макетам 24.dp

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/onboarding/OnboardingScreen.kt:
```diff
@@ -0,0 +1,124 @@
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(horizontal = VolleyDimens.DIMEN_28.dp),
```
Кажется, тут 26.dp размер должен быть

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/onboarding/OnboardingScreen.kt:
```diff
@@ -0,0 +1,124 @@
+                    contentDescription = null,
+                    modifier = Modifier.size(VolleyDimens.DIMEN_200.dp)
+                )
+                Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_16.dp))
```
Здесь тоже отступ 24.dp

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/registration/RegistrationScreen.kt:
```diff
@@ -0,0 +1,346 @@
+            .fillMaxSize()
+            .background(VolleyColor.TurquoiseDark)
+    ) {
+        VolleyContainersRootTransparent.GlassContainer(
```
А здесь точно GlassContainer нужен? Мне кажется, там эффект стекла только по краям. А Glass размытие по всему контейнеру даёт

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/registration/RegistrationScreen.kt:
```diff
@@ -0,0 +1,346 @@
+                    top = VolleyDimens.DIMEN_8.dp,
+                    bottom = VolleyDimens.DIMEN_4.dp
+                ),
+            cornerRadius = VolleyDimens.DIMEN_24,
```
Кажется, скругления 32.dp

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/registration/RegistrationScreen.kt:
```diff
@@ -0,0 +1,346 @@
+                    color = VolleyColor.White
+                )
+
+                Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_8.dp))
```
После поля регистрации отступ на макетах 16.dp

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/registration/RegistrationScreen.kt:
```diff
@@ -0,0 +1,346 @@
+                )
+
+                Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_16.dp))
+                Box(
```
Для разделителя можно использовать инструмент HorizontalDivider, не обязательно всё в Box делать)

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/registration/RegistrationScreen.kt:
```diff
@@ -0,0 +1,346 @@
+                )
+
+                Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_16.dp))
+                Box(
```
Аналогично повторяющийся элемент можно вынести в отдельный compose

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/registration/RegistrationScreen.kt:
```diff
@@ -0,0 +1,346 @@
+                )
+
+                Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_16.dp))
+                Box(
```
И тут тоже) Box лучше вынести в отдельный compose или сделать divider с одинаковыми padding по 16.dp и везде его понаставлять. тогда и Spacer уйдут и код станет меньше и читабельнее

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/registration/RegistrationScreen.kt:
```diff
@@ -0,0 +1,346 @@
+                    contentAlignment = Alignment.CenterStart
+                ) {
+                    VolleyText.BodyRegular(
+                        text = if (state.country.isEmpty()) {
```
Здесь можно заменить на ifEmpty, студия наверное подскажет

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/registration/RegistrationScreen.kt:
```diff
@@ -0,0 +1,346 @@
+                    )
+                }
+
+                Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_24.dp))
```
Здесь до кнопочки 16.dp

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/registration/RegistrationScreen.kt:
```diff
@@ -0,0 +1,346 @@
+    )
+}
+
+@Preview(showBackground = true, showSystemUi = true)
```
Можно растянуть preview, добавив параметр 
heightDp = 1000

Evlenium replied:
Можно растянуть preview, добавив параметр 
heightDp = 1000

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/registration/RegistrationScreen.kt:
```diff
@@ -0,0 +1,346 @@
+}
+
+@Composable
+private fun GroupButtonsForLevel(
```
GroupButtonsForGender2 и GroupButtonsForLevel похоже не используются

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/signup/SignUpScreen.kt:
```diff
@@ -0,0 +1,179 @@
+                    start = VolleyDimens.DIMEN_16.dp,
+                    end = VolleyDimens.DIMEN_16.dp,
+                    top = VolleyDimens.DIMEN_20.dp,
+                    bottom = VolleyDimens.DIMEN_20.dp
```
Можно смело заменять на horizontal и vertical

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/signup/SignUpScreen.kt:
```diff
@@ -0,0 +1,179 @@
+                )
+            }
+
+            Spacer(modifier = Modifier.height(VolleyDimens.DIMEN_12.dp))
```
Отступы между кнопками на макетах 8.dp, сделай одну кастомную кнопку на compose, которой будешь присваивать значения текста, иконки и цвета. А то принцип DRY нарушается.

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/registration/RegistrationScreen.kt:
```diff
@@ -0,0 +1,346 @@
+                    contentAlignment = Alignment.CenterStart
+                ) {
+                    VolleyText.BodyRegular(
+                        text = if (state.city.isEmpty()) stringResource(id = R.string.your_city) else state.city,
```
Здесь тоже сработает ifEmpty конструкция, студия предложит

Evlenium reviewed (commented):
Привет Дим! В целом, всё хорошо, за исключением некоторых несостыковок с макетами и повторений кода, иногда нарушается принцип DRY. Многое можно переиспользовать в compose, задав дополнительные поля. В общем, посмотри по макетам. Жду новых правок! =)

Commit:
Merge branch 'develop' into 61-screen-flow-authorization

# Conflicts:
#	app/src/main/java/cy/volleybolley/core/presentation/ui/model/VolleyDimens.kt

Commit:
#71 1-я Часть правок по ревью
1. Удалены лишние отступу
2. Поправлены размеры в соотвествии с дизайном
3. Вынос рекомендуемых компонентов в отдельные библиотечные функции
4. Библиотечный градиент с багом, пока что замокано на двустрочные элементы заканчивать градиент на первой
5. Использован библиотечный HorizontalDivider вместо самописного решения
6. Заменено if else, на ifEmpty в соответсвии с kotlin style
7. Удалены неиспользуемые функции и файлы

Secur1ty-21 reviewed (commented):
Осталось:
1. Рефакторинг RegistrationScreen
2. Верстка RegistrationByPhoneScreen

Commit:
#71 2-я часть правок по пр, рефакторинг RegistraionScreen

Commit:
#61 Добавлены экраны подтверждения номера телефона, добавлены провекри на заполненность экрана регистрации

Commit:
Merge branch 'develop' into 61-screen-flow-authorization

# Conflicts:
#	app/src/main/java/cy/volleybolley/core/DiProvider.kt

Evlenium commented on app/src/main/java/cy/volleybolley/core/domain/VolleyFeature.kt:
```diff
@@ -0,0 +1,5 @@
+
+object VolleyFeature {
+    const val IS_AUTH_BY_FACEBOOK_AVAILABLE = false
+}
```
А эта константа точно должна быть в core.domain слое? Вроде как она фигурирует в presentation

Evlenium replied:
Может тогда имеет смысл добавить все регистрации, вдруг от какой-то откажемся?)

Secur1ty-21 replied:
Да, константа точно должна быть в этом слое. Соглашусь с добавлением остальных авторизаций сюда

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/component/VolleyButton.kt:
```diff
@@ -1122,11 +1129,76 @@ object VolleyButton {
+            modifier = modifier.height(VolleyDimens.DIMEN_56.dp),
+            colors = ButtonDefaults.buttonColors(containerColor = backgroundColor),
+            shape = RoundedCornerShape(VolleyDimens.DIMEN_16.dp),
+            contentPadding = PaddingValues(all = 16.dp)
```
Кажется, у нас в ресурсах есть VolleyDimens.DIMEN_16. Может стоит им воспользоваться?

Secur1ty-21 replied:
Касательно всех замечаний по поводу выноса размеров в ресурсы:
Их лучше выносить осмысленно, например, по дизайну у всех видов кнопок одинаковая высота, поэтому выносим этот размер, чтобы в будущем, если вдруг дизайнеры решат увеличить размер кнопок по фидбеку пользователей, то можно было это сделать в одном месте. Сейчас если захотим через эти ресурсы что-то поменять, непонятно что это затронет, даже видел использование этих ресурсов в обычной логике кода, не касающейся ui у нас на проекте

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/component/VolleyButton.kt:
```diff
@@ -1122,11 +1129,76 @@ object VolleyButton {
+                modifier = Modifier.size(VolleyDimens.DIMEN_24.dp)
+            )
+            VolleyText.BodyBoldMedium(
+                modifier = Modifier.padding(start = 12.dp),
```
И здесь как будто не хватает выноса в ресурсы)

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/component/VolleyButton.kt:
```diff
@@ -1122,11 +1129,76 @@ object VolleyButton {
+        onClick: () -> Unit
+    ) {
+        TextButton(
+            modifier = modifier.height(18.dp),
```
Здесь тоже можно воспользоваться ресурсами

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/navigation/NavHostContainer.kt:
```diff
@@ -55,23 +56,62 @@ import cy.volleybolley.core.presentation.ui.screens.profile.ProfileScreen
+                paddingFromSystemUi = paddingFromSystemUi
+            )
+        }
+        composable<SignUpRoute> {
```
Здесь скорее к неймингу вопросы, мне кажется, что на этом экране всё-таки пользователь выполняет вход нежели регистрацию. Я бы переименовал в "LoginRoute" или в "SignInRoute". А то у нас получается 2 экрана регистрации, просто по разному называются.

Secur1ty-21 replied:
У нас по логике по телефону или гуглу может быть как вход так и первичная регистрация, соглашусь что signUp не подходит, переименую в Authorization

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/aboutlevels/AboutLevelsViewModel.kt:
```diff
@@ -0,0 +1,23 @@
+    override fun obtainEvent(event: AboutLevelsEvent) {
+        when (event) {
+            AboutLevelsEvent.OnBackClicked -> {
+                viewModelScope.launch {
```
Может быть имеет смысл указать Dispatcher.Main явно, а то когда всякое во viewModel добавится - то можно будет запутаться

Secur1ty-21 replied:
Оно вообще не используется, забыл удалить

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/aboutlevels/AboutLevelsScreen.kt:
```diff
@@ -0,0 +1,127 @@
+                    prefixResId = R.string.level_hard_prefix,
+                    descriptionResId = R.string.level_hard_description,
+                    gradient = Brush.verticalGradient(
+                        Pair(first = 0f, second = VolleyColor.YellowForGradient),
```
Как-то странно на прошлом levelItem с одной стилистикой написано:
                        0f to VolleyColor.YellowForGradient,
                        1f to VolleyColor.GreenForGradient,
 А на этом с другой

Secur1ty-21 replied:
В этом коммите писал почему так, п.4 https://github.com/o1isrta/VolleyBolleyAndroid/pull/71/commits/f0cc7d56b2beb960d0e653ddb13d8031e19c4f79

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/aboutlevels/AboutLevelsScreen.kt:
```diff
@@ -0,0 +1,127 @@
+                    gradient = Brush.verticalGradient(
+                        Pair(first = 0f, second = VolleyColor.YellowForGradient),
+                        Pair(first = 0.49f, second = VolleyColor.GreenForGradient)
+                    )
```
Я бы наверное LevelItem's сделал через forEachIndexed с аргументам через data class. Но это скорее вкусовщина)

Evlenium replied:
Хотя у меня тоже был gradient и не работал, а с листом Items заработал

Secur1ty-21 replied:
Я наоборот предпочитаю плодить как можно меньше сущностей, поэтому пока оставлю так. Твой вариант лучше масштабируется, но думаю пока что можно оставить так

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/launch/LaunchScreen.kt:
```diff
@@ -0,0 +1,101 @@
+            modifier = Modifier.size(VolleyDimens.LAUNCH_LOGO_SIZE.dp)
+        )
+        VolleyText.LogoDisplay(
+            modifier = Modifier.padding(top = 24.dp),
```
Здесь можно воспользоваться ресурсами

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/launch/LaunchScreen.kt:
```diff
@@ -0,0 +1,101 @@
+        )
+        if (isProgressBarVisible) {
+            VolleyProgress.CircularProgress(modifier = Modifier.padding(top = 16.dp))
+        }
```
И здесь тоже ресурсы)

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/onboarding/OnboardingScreen.kt:
```diff
@@ -0,0 +1,93 @@
+                color = VolleyColor.White
+            )
+            VolleyText.HeroBody(
+                modifier = Modifier.padding(top = 16.dp, start = 26.dp, end = 32.dp),
```
Здесь в ресурсах тоже есть размеры 16.dp, 26, 32 и 50 тоже есть. Можно вынести)

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/launch/LaunchScreen.kt:
```diff
@@ -0,0 +1,101 @@
+                }
+            }
+
+            else -> {}
```
А мы не null тут передаём?

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/registration/RegistrationScreen.kt:
```diff
@@ -0,0 +1,318 @@
+
+        item {
+            VolleyTextFieldGradient.GradientSpinner(
+                modifier = Modifier.padding(top = 16.dp, bottom = 188.dp).fillMaxWidth(),
```
Здесь, кажется, когда скроллишь маленький экран скролл чуть с запасом уходит вниз.
<img width="488" height="950" alt="image" src="https://github.com/user-attachments/assets/04e2f287-4ad2-4341-ac5c-a39df6443d57" />

Secur1ty-21 replied:
Все верно, это специально сделано чтобы была возможность раскрыть dropDown как показано в дизайне, иначе он отрисуется сверху (dropDown должен рисоваться поверх элементов)

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/registration/RegistrationScreen.kt:
```diff
@@ -0,0 +1,318 @@
+            onSelected = onLevelClick
+        )
+        HorizontalDivider(
+            modifier = Modifier.padding(top = 16.dp),
```
В целом всё хорошо, только VolleyDimens не используются на этом экране совсем и при скролле паддинг большой

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/signup/SignUpScreen.kt:
```diff
@@ -0,0 +1,176 @@
+            )
+        }
+        VolleyButton.ActiveButtonWithLeadingIcon(
+            modifier = Modifier.padding(top = 8.dp).fillMaxWidth(),
```
VolleyDimens.DIMEN_8?

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/authorizationByPhone/sendCode/presentation/AuthorizationByPhoneScreen.kt:
```diff
@@ -0,0 +1,106 @@
+    VolleyContainersRootTransparent.TransparentContainer(
+        modifier = Modifier
+            .padding(top = paddingFromSystemUi.calculateTopPadding() + 8.dp, start = 8.dp, end = 8.dp)
+            .fillMaxWidth()
```
Здесь тоже к экрану можно применить VolleyDimens.DIMEN_8 и другие..

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/authorizationByPhone/verifyCode/presentation/VerifyPhoneNumberScreen.kt:
```diff
@@ -0,0 +1,122 @@
+) {
+    VolleyContainersRootTransparent.TransparentContainer(
+        modifier = Modifier
+            .padding(top = paddingFromSystemUi.calculateTopPadding() + 8.dp, start = 8.dp, end = 8.dp)
```
Тоже здесь все размеры есть в ресурсах)

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/signupByPhone/sendCode/presentation/RegistrationByPhoneViewModel.kt:
```diff
@@ -0,0 +1,46 @@
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.update
+
+class RegistrationByPhoneViewModel : BaseViewModel<SignupByPhoneState, SignupByPhoneEvent, SignupByPhoneEffect>(
```
Здесь будто бы не хватает метода, который бы вообще верифицировал бы введённые данные телефона перед отправкой на сервер. Ну, то есть, у нас есть Bollean поле с ошибкой валидации, но на макетах есть конкретные ошибки с message. А тут всегда message один получается.

Secur1ty-21 replied:
Согласен, планировал дальше добавлять как приступлю к свзяке с сетевым слоем авторизации

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/authorizationByPhone/sendCode/presentation/model/AuthorizationByPhoneState.kt:
```diff
@@ -0,0 +1,10 @@
+    val isLoading: Boolean = false,
+    val phoneNumber: String = "",
+    val isPhoneNumberInputError: Boolean = false,
+    val isBtnSendCodeEnabled: Boolean = false
```
Я бы добавил конкретное поле errorMessage: String, в котором бы указывал, слишком короткий телефонный номер/длинный или неправильный

Secur1ty-21 replied:
Добавлю позже, вместе с сетевым слоем

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/signupByPhone/verifyCode/presentation/VerifyPhoneNumberScreen.kt:
```diff
@@ -0,0 +1,122 @@
+                VolleyButton.GradientTextButton(
+                    modifier = Modifier.padding(top = 8.dp).align(Alignment.CenterHorizontally),
+                    text = if (state.isBtnNewCodeEnabled) {
+                        stringResource(R.string.send_code)
```
Здесь по макетам идёт не "Send code", а "Get new code". И он зелёненький градиент в жёлтый становится, после remaining Time

Secur1ty-21 replied:
С градиентом вроде все ок, надпись поправил

Evlenium reviewed (commented):
Код огонь! Многое для себя подчеркнул нового!
Есть небольшой вопрос с отступами на экране регистрации. Ну и кажется по макетам button чуть другого стиля и содержит другое сообщение, ну и полная верификация телефона отсутствует с полями слишком длинно, слишком коротко и некорректно.
Наши VolleyDimmens немножко проигнорены)

Commit:
#71 Правки по пр
1. Нейминг экрана выбора способа авторизации и авторизации по номеру телефона
2. Добавление feature-toggle для авторизаций по номеру телефона и google
3. Удалены неиспользуемые файлы
4. Исправлен текст кнопки отправки нового кода, на экране подвтерждения номера телефона в соответсвии с макетом

Evlenium commented on app/src/main/java/cy/volleybolley/core/presentation/ui/screens/authorization/authorizationByPhone/verifyCode/presentation/VerifyPhoneNumberScreen.kt:
```diff
@@ -82,7 +82,7 @@ fun VerifyPhoneNumberScreen(
                 VolleyButton.GradientTextButton(
                     modifier = Modifier.padding(top = 8.dp).align(Alignment.CenterHorizontally),
```
Вот тут получилось не совсем по макетам, на макетах эта кнопочка как и текст сделаны по центру экрана. А тут получается, что не по центру, а по левому краю
<img width="412" height="250" alt="image" src="https://github.com/user-attachments/assets/9112b6c9-5a9f-487c-ac66-22ca7d1a4c0c" />

Evlenium commented on app/src/main/java/cy/volleybolley/core/DiProvider.kt:
```diff
@@ -20,8 +20,8 @@ object DiProvider {
         profileModule,
         launchModule,
-        signUpModule,
+        AuthorizationModule,
```
Наверное тут camelCase лучше использовать для di

Evlenium commented:
Перепроверил ещё разок ПР. Оставил некоторые замечания:
1) Там одна из view, как раз та градиентная кнопка newCode на макетах выровнена по центру, а у нас получилось так, что она осталась с левой стороны.
2) В di надо бы чуть подправить нейминг.
3) Раз уж мы используем в эффектах null -> {} - то не мешало бы поправить и такие экраны как:
VerifyPhoneNumberScreen,
RegistrationScreen,
AuthorizationScreen.

Commit:
#71 Правки по пр
1. Нейминг модуля
2. Выравнивание кнопки получения нового кода на экране подтверждения номера телефона по центру
3. В when вместо ветки else перечисление всех вариантов

Evlenium reviewed (approved):
Всё работает! Ждём реализации сетевых слоёв регистрации
